spot_wrapper/calibration/calibration_util.py

```diff
@@ -0,0 +1,935 @@
+
+    Returns:
+        Optional[Tuple[np.ndarray, np.ndarray]]: The rotation vector and translation vector
+        representing the 6D pose of the checkerboard center if found, else None.
+    """
+    charuco_corners, charuco_ids = detect_charuco_corners(img, charuco_board, aruco_dict)
+
+    if charuco_corners is not None and charuco_ids is not None:
+        # Estimate the pose of the Charuco board
+        rvec = np.zeros((3, 1))
+        tvec = np.zeros((3, 1))
+        retval, rvec, tvec = cv2.aruco.estimatePoseCharucoBoard(
+            charuco_corners,
+            charuco_ids,
+            charuco_board,
+            camera_matrix,
+            dist_coeffs,
+            rvec,
+            tvec,
+        )
+        if retval:
+            # Compute the translations needed to transform to the center
+            x_trans = (charuco_board.getSquareLength() * charuco_board.getChessboardSize()[0]) / 2.0
+            y_trans = (charuco_board.getSquareLength() * charuco_board.getChessboardSize()[1]) / 2.0
+            # Adjust tvec to be relative to the center
+            center_trans = np.array([x_trans, y_trans, 0.0]).reshape((3, 1))
+            rmat, _ = cv2.Rodrigues(rvec)
+            tvec = tvec + rmat.dot(center_trans)
+
+            return np.array(rmat), np.array(tvec).ravel()
+        else:
+            raise ValueError(
+                "Corners were found, but failed to localize. You likely primed the robot too close to the board"
+            )
+    else:
+        raise ValueError(
+            "Couldn't detect any Charuco Boards in the image, "
+            "localization failed. Ensure the board is visible from the"
+            " primed pose."
+        )
+
+
+def convert_camera_t_viewpoint_to_origin_t_planning_frame(
+    origin_t_planning_frame: np.ndarray = np.eye(4),
+    planning_frame_t_opencv_camera: np.ndarray = np.eye(4),
+    opencv_camera_t_viewpoint: np.ndarray = np.eye(4),
+) -> np.ndarray:
+    """
+    Convert a camera viewpoint given in a camera frame, to the planning frame.
+
+    Camera frame should be in OpenCV/ ROS format, where
+    +x should point to the right in the image
+    +y should point down in the image
+    +z should point into to plane of the image.
+
+    Assuming that origin_t_planning_frame isn't aligned with this format, align it
+    so that the viewpoint can be localized as a hand pose command.
+
+    Args:
+        origin_t_planning_frame (np.ndarray, optional): 4x4 homogenous transform
+            from origin to planning frame. Defaults to np.eye(4).
+        planning_frame_t_opencv_camera (np.ndarray, optional): 4x4 homogenous transform
+            from planning frame to the camera. Defaults to np.eye(4).
+        opencv_camera_t_viewpoint (np.ndarray, optional): 4x4 homogenous transform
+            from camera to the future viewpoint. Defaults to np.eye(4).
+
+    Returns:
+        np.ndarray: 4x4 homogenous transform of origin to planning frame of future viewpoint.
```

Comment:
nit: the origin_t_planning_frame_next is technically worded the other way planning frame of future view point to origin. Its a bit counter intuitive but have to read it backward given our style guide. This is taking a point in the planning frame and moving it to the origin frame.

Reply:
changed to 


```np.ndarray: 4x4 homogenous transform of the planning frame of a future viewpoint
            expressed in the origin frame.```